options.go

```diff
@@ -103,3 +107,22 @@ func (h headerSetter) withExporter(e *Exporter) {
 func WithHeaders(headers map[string]string) ExporterOption {
 	return headerSetter(headers)
 }
+
+type clientCredentials struct {
+	credentials.TransportCredentials
+}
+
+var _ ExporterOption = (*clientCredentials)(nil)
+
+// WithTLSCredentials allows the connection to use TLS credentials
+// when talking to the server. It takes in grpc.TransportCredentials instead
+// of say a Certificate file or a tls.Certificate, because the retrieving
+// these credentials can be done in many ways e.g. plain file, in code tls.Config
+// or by certificate rotation, so it is up to the caller to decide what to use.
+func WithTLSCredentials(creds credentials.TransportCredentials) ExporterOption {
+	return &clientCredentials{TransportCredentials: creds}
```

Comment:
nit: consider adding a check that `WithInsecure` and `WithTLSCredentials` shouldn't be called at the same time.

Reply:
Oh, if you see in New, I do that
https://github.com/census-ecosystem/opencensus-go-exporter-ocagent/blob/5a6e73f487e155ca74647bf2acdcf88e09133e2a/ocagent.go#L263-L267

Reply:
so they are mutually exclusive!